addons/mail/static/tests/chatter_tests.js

```diff
@@ -291,6 +302,79 @@ QUnit.test('Activity Done keep feedback on blur', function (assert) {
     });
 });
 
+QUnit.test('Activity Done by uploading a file', function (assert) {
+    assert.expect(4);
+
+    // simulate (shortcut) the upload and trigger the event when the
+    // attachment is created serverside.
+    testUtils.mock.patch(Activity, {
+        _onMarkActivityDoneUploadFile: function (ev) {
+            var $markDoneBtn = $(ev.currentTarget);
+            var fileuploadID = $markDoneBtn.data('fileupload-id');
+
+            var win = window.top.window;
+            win.jQuery(win).trigger(fileuploadID, [{
+                id:3, name:"name2",
+                type:'binary',
+                mimetype:"application/x-msdos-program",
+                datas_fname:"file2.txt",
+                res_id: 5,
+                res_model: 'partner'
+            }]);
+        },
+    });
+
+    // create the activity
+    this.data['mail.activity'].records = [{
+        activity_type_id: 3,
+        id: 1,
+        can_write: true,
+        user_id: 2,
+        state: 'today',
+        note: 'But I\'m talkin\' about Shaft',
```

Comment:
UI strings are usually double quoted `"`, whereas technical strings are single quoted `'`. Here you can use double quotes to prevent single quote escaping.